Update to netstandard2.0.

fernando-almeida opened the pull request:
Updated all external package dependencies to target latest versions.
Revised target classlib framework to netstandard2.0.
Revised obsolete properties/methods in external packages.
Revised msg pack serialization test.

megakid commented on Obvs.Serialization.MessagePack-CSharp/Obvs.Serialization.MessagePack-CSharp.csproj:
```diff
@@ -10,8 +10,8 @@
   <ItemGroup>
-    <PackageReference Include="MessagePack" Version="1.7.3.2" />
-    <PackageReference Include="Obvs" Version="4.1.0.9" />
+    <PackageReference Include="MessagePack" Version="1.7.3" />
```
Any reason for this downgrade?  I see `MessagePack 1.7.3.4` is out ? https://www.nuget.org/packages/MessagePack/

megakid commented:
> No real reason for having done that. It’s probably due to a typo I have made.
> […](#)
> On 19 Dec 2018, at 10:57, James Connor ***@***.***> wrote: @megakid commented on this pull request. In Obvs.Serialization.MessagePack-CSharp/Obvs.Serialization.MessagePack-CSharp.csproj <[#9 (comment)](https://github.com/christopherread/Obvs.Serialization/pull/9#discussion_r242874017)>: > @@ -10,8 +10,8 @@ </PropertyGroup> <ItemGroup> - <PackageReference Include="MessagePack" Version="1.7.3.2" /> - <PackageReference Include="Obvs" Version="4.1.0.9" /> + <PackageReference Include="MessagePack" Version="1.7.3" /> Any reason for this downgrade? I see MessagePack 1.7.3.4 is out ? https://www.nuget.org/packages/MessagePack/ <https://www.nuget.org/packages/MessagePack/> — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <[#9 (review)](https://github.com/christopherread/Obvs.Serialization/pull/9#pullrequestreview-186490377)>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4mtq2JUkae940LsqgdpQwa6Bc9Fd4oks5u6hu0gaJpZM4ZZe7H>.

Ok I have fixed that.

fernando-almeida commented:
I’ve update the pull request and fixed any outstanding conflicts.
I find it very tricky so far to get stuff working with both .NET framework (using Mono) and .NET core.
The latter works pretty good, but the 1st one has a lot of nooks & crannies.
I was able to successfully build with .NET Framework 4.5 (I’ve considered a further downgrade since your last package actually targets this).
Targeting netstandard 2.0 will automatically include .NET Framework 4.6.1 (but not the other way around) as depicted here https://immo.landwerth.net/netstandard-versions/# <https://immo.landwerth.net/netstandard-versions/#> so this should be the preferred target going forward for cross-platform support.


> On 19 Dec 2018, at 11:37, James Connor <notifications@github.com> wrote:
> 
> No real reason for having done that. It’s probably due to a typo I have made.
> … <x-msg://7/#>
> On 19 Dec 2018, at 10:57, James Connor ***@***.***> wrote: @megakid <https://github.com/megakid> commented on this pull request. In Obvs.Serialization.MessagePack-CSharp/Obvs.Serialization.MessagePack-CSharp.csproj <#9 (comment) <https://github.com/christopherread/Obvs.Serialization/pull/9#discussion_r242874017>>: > @@ -10,8 +10,8 @@ - - + Any reason for this downgrade? I see MessagePack 1.7.3.4 is out ? https://www.nuget.org/packages/MessagePack/ <https://www.nuget.org/packages/MessagePack/> https://www.nuget.org/packages/MessagePack/ <https://www.nuget.org/packages/MessagePack/> — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <#9 (review) <https://github.com/christopherread/Obvs.Serialization/pull/9#pullrequestreview-186490377>>, or mute the thread https://github.com/notifications/unsubscribe-auth/AB4mtq2JUkae940LsqgdpQwa6Bc9Fd4oks5u6hu0gaJpZM4ZZe7H <https://github.com/notifications/unsubscribe-auth/AB4mtq2JUkae940LsqgdpQwa6Bc9Fd4oks5u6hu0gaJpZM4ZZe7H>.
> 
> Ok I have fixed that.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/christopherread/Obvs.Serialization/pull/9#issuecomment-448564719>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4mtmtmMAMIEyGCemp-MWeU0mQQ0DC6ks5u6iUXgaJpZM4ZZe7H>.
>

fernando-almeida commented:
I could probably need your help to debug the error that's being thrown.
I added the xunit console runner to try running the tests on my MacOS but apparently Xunit only supports .NET core on Linux/MacOS. I'm able to run the tests fine if a specify the framework netcoreapp2.0 but am unable to use net45 which yielded the error that is shown on appveyor.

megakid commented:
I'll take a look on my VS2017 windows machine and resolve any issues.  Mono hasn't been tested (and as you say, shouldn't be needed with `.netstandard`) but I understand you have a Mac so won't be able to test the full framework easily!

megakid commented:
Also see https://twitter.com/terrajobst/status/1031999730320986112?lang=en - recommended to use 4.7.2 for proper .netstandard2.0 support on full framework - it's not recommended to use the .netstandard2.0 builds on anything less than that.

fernando-almeida commented:
Ok, thanks James.
The interactive chart I linked seems very clear.
Let me know what you found to check if it's something I can fix in my MacOS
to fully run the tests. I can build fine and execute the binaries with Mono
however running Xunit tests does not seem to work.

On Wed, Dec 19, 2018, 13:49 James Connor <notifications@github.com wrote:

> Also see https://twitter.com/terrajobst/status/1031999730320986112?lang=en
> - recommended to use 4.7.2 for proper .netstandard2.0 support on full
> framework - it's not recommended to use the .netstandard2.0 builds on
> anything less than that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/christopherread/Obvs.Serialization/pull/9#issuecomment-448602630>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB4mtnCuEPi2VtHMD9s2S1zFg-JmXarMks5u6kPfgaJpZM4ZZe7H>
> .
>
